src/main/kotlin/persistence/mapping/UserMapper.kt

```diff
@@ -0,0 +1,48 @@
+package persistence.mapping
+
+import data.Language
+import data.User
+import data.dao.Dao
+import data.mapping.Mapper
+import persistence.model.*
+import persistence.repo.UserLanguageRepo
+import javax.inject.Inject
+
+class UserMapper @Inject constructor(private val userLanguageRepo: UserLanguageRepo, private val languageRepo: Dao<Language>): Mapper<IUserEntity, User> {
+    private val userPreferencesMapper = UserPreferencesMapper(languageRepo)
+    /**
+     * takes a User object and maps and returns a IUserEntity
+     */
+    override fun mapToEntity(type: User): IUserEntity {
+        val userEntity = UserEntity()
+        userEntity.id = type.id
+        userEntity.setAudioHash(type.audioHash)
+        userEntity.setAudioPath(type.audioPath)
+        userEntity.setUserPreferencesEntity(userPreferencesMapper.mapToEntity(type.userPreferences))
+        return userEntity
+    }
+
+    /**
+     * Takes a IUserEntity and maps and returns a User object
+     */
+    override fun mapFromEntity(type: IUserEntity): User {
+        // queries to find all the source languages
+        val allUserLanguageJunctionTableRows = userLanguageRepo.getByUserId(type.id).blockingFirst()
```

Comment:
so like; this could just be userLanguages?
allUserHolyImplementationDetailsBatman ends up killing your line length and is difficult to understand at a quick glance